addons/openebs/template/base/install.sh

```diff
@@ -48,13 +48,56 @@ function openebs() {
 function openebs_maybe_migrate_from_rook() {
     if [ -z "$ROOK_VERSION" ]; then
         if kubectl get ns | grep -q rook-ceph; then
-            rook_ceph_to_sc_migration "$OPENEBS_LOCALPV_STORAGE_CLASS"
+            # show validation errors from pvmigrate
+            # if there are errors, openebs_maybe_rook_migration_checks() will bail
+            openebs_maybe_rook_migration_checks
+            rook_ceph_to_sc_migration "$OPENEBS_LOCALPV_STORAGE_CLASS" "1"
             DID_MIGRATE_ROOK_PVCS=1 # used to automatically delete rook-ceph if object store data was also migrated
         fi
     fi
 }
 
-function prompt_migrate_from_rook() {
+function openebs_maybe_rook_migration_checks() {
+
+    # get the list of StorageClasses that use rook-ceph
+    local rook_scs
+    local rook_default_sc
+    rook_scs=$(kubectl get storageclass | grep rook | grep -v '(default)' | awk '{ print $1}') # any non-default rook StorageClasses
+    rook_default_sc=$(kubectl get storageclass | grep rook | grep '(default)' | awk '{ print $1}') # any default rook StorageClasses
+
+    # Ensure openebs-localpv-provisioner deployment is ready
+    echo "awaiting openebs-localpv-provisioner deployment"
+    spinner_until 120 deployment_fully_updated openebs openebs-localpv-provisioner
+
+    echo "running Rook to OpenEBS migration checks ..."
+    local rook_scs_pvmigrate_dryrun_output
+    local rook_default_sc_pvmigrate_dryrun_output
+    for rook_sc in $rook_scs
+    do
+        # run validation checks for non default Rook storage classes
+        rook_scs_pvmigrate_dryrun_output=$($BIN_PVMIGRATE --source-sc "$rook_sc" --dest-sc "$OPENEBS_LOCALPV_STORAGE_CLASS" --rsync-image "$KURL_UTIL_IMAGE" --preflight-validation-only || true)
+        if [ $? -ne 0 ]; then
+            break
+        fi
```

Comment:
i think this will work but please test it

```suggestion
        if ! rook_scs_pvmigrate_dryrun_output=$($BIN_PVMIGRATE --source-sc "$rook_sc" --dest-sc "$OPENEBS_LOCALPV_STORAGE_CLASS" --rsync-image "$KURL_UTIL_IMAGE" --preflight-validation-only) ; then
            break
        fi
```

Reply:
This works!